syntax = "proto3";
package monitor.proto;

import "google/protobuf/empty.proto";


message SoftIrq {
  string cpu = 1;
  float hi = 2;
  float timer = 3;
  float net_tx = 4;
  float net_rx = 5;
  float block = 6;
  float irq_poll = 7;
  float tasklet = 8;
  float sched = 9;
  float hrtimer = 10;
  float rcu = 11;
}


message CpuLoad {
  float load_avg_1 = 1;
  float load_avg_3 = 2;
  float load_avg_15 = 3;
}

message CpuStat{
  string cpu_name = 1;
  float cpu_percent = 2;
  float usr_percent = 3;
  float system_percent = 4;

}

message MonitorInfo{
  repeated SoftIrq soft_irq = 4;
  CpuLoad cpu_load = 5;
  repeated CpuStat cpu_stat = 6;
}

service GrpcManager {
  rpc SetMonitorInfo(MonitorInfo) returns (google.protobuf.Empty) {
  }

  rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo) {
  }
}
